airbyte-webapp/src/views/Connection/ConnectionForm/calculateInitialCatalog.test.ts

```diff
@@ -391,4 +388,151 @@ describe("calculateInitialCatalog", () => {
     expect(streams[1]).toHaveProperty("config.cursorField", ["selected_path"]);
     expect(streams[2]).toHaveProperty("config.cursorField", []);
   });
+
+  test("source defined properties should override the saved properties", () => {
+    const { stream, config } = mockSyncSchemaStream;
+
+    const { streams } = calculateInitialCatalog(
```

Comment:
nit: I think it would slightly help the readability of these tests if these variables were renamed to something like `{ mockStream, mockConfig }` or `{ startingStream, startingConfig }`, and like `{ resultStreams}`. I think this would make the `expect()` statements at the bottom slightly easier to parse.

But other than that these tests look great!